scanner_cli.py

```diff
@@ -245,23 +301,93 @@ def store_user_data(username):
 
         print(f"User data for {username} has been updated.")
 
+def calculate_metrics(property_id, state_code, city):
+    # Load data from JSON files
+    with open("median_price_by_region.json", 'r', encoding='utf-8') as price_file:
+        median_price_data = json.load(price_file)
+
+    with open("median_rent_by_region.json", 'r', encoding='utf-8') as rent_file:
+        median_rent_data = json.load(rent_file)
+
+    # Get the specific fetched real estate file
+    properties_file = "real_estate_for_sale_" + city + "_" + state_code + ".json"
+    with open(properties_file, 'r', encoding='utf-8') as real_estate_file:
+        real_estate_data = json.load(real_estate_file)
+
+    # Mortgage Rates
+    with open("us_mortgage_rate.json", 'r', encoding='utf-8') as mortgage_file:
+        mortgage_rate_data = json.load(mortgage_file)
+
+    # Find the property in real_estate_data using property_id
+    property_info = None
+    for info in real_estate_data:
+        if info['property_id'] == property_id:
+            property_info = info
+            break
+
+    # Extract state and city from the real estate data
+    state = property_info['address']['state_code']
+    city = property_info['address']['city']
+
+    # Find the median price based on state and city
+    median_price = None
+    for state_data in median_price_data:
```

Comment:
is this median_price_data a dict? or array? Look into using https://docs.python.org/3/library/typing.html